File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter09/XMLFormatter/Receiver/ReceiverMain.cs
using System;
using System.Messaging;
using CustomerLibrary;

namespace XmlReceiver
{
   class XmlReceiverMain
   {
      static void Main(string[] args)
      {
         // Open queue
         MessageQueue mq = new MessageQueue(@".\private$\MyPrivateQ");
      
         // read the message
         Message msg = mq.Receive();

         // Get the message type from the label
         string[] expectedTypeNames = new String[] {msg.Label};     

         // Set the formatter object for the message
         msg.Formatter = new XmlMessageFormatter(expectedTypeNames);

         // Retrieve the message
         object o = msg.Body;

         // process the message ...
         Console.WriteLine(o.ToString());
      }
   }
}

File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter09/XmlFormatter/Receiver/ReceiverMain.vb
Imports System.Messaging
Imports System.Threading

Module ReceiverMain

   Sub Main()
      'Open queue
      Dim mq As New MessageQueue(".\private$\MyPrivateQ")
	
      'Receive message
      Dim msg As Message = mq.Receive()
            
      'Get the message type from the label, and set expected type names
      Dim expectedTypeNames() As String = New String() {msg.Label}   

      'Set the formatter object for deserializing this message
      msg.Formatter = new XmlMessageFormatter(expectedTypeNames)

      'Retrieve the message body
      Dim o As Object = msg.Body

      'Use Customer object in a late bound fashion ...
      
      'Process Customer data     
      Console.WriteLine(o.ToString())
   End Sub
   
End Module